specs/lightclient/validator.md

```diff
@@ -0,0 +1,381 @@
+
+#### Sync committee signatures
+
+##### Prepare sync committee signature
+
+If a validator is in the current sync committee (i.e. `get_sync_committee_assignment` above returns an index for the current epoch in question), then for every slot in the current sync committee period the validator should prepare a `SyncCommitteeSignature` according to the logic in `get_sync_committee_signatures` as soon as they have determined the head block of the current slot. Note that if a validator is duplicated in the sync committee, they should broadcast multiple signatures. 
+
+This logic is triggered upon the same conditions as when producing an attestation. Meaning, a sync committee member should produce and broadcast a `SyncCommitteeSignature` either when (a) the validator has received a valid block from the expected block proposer for the current `slot` or (b) one-third of the slot has transpired (`SECONDS_PER_SLOT / 3` seconds after the start of the slot) -- whichever comes first.
+
+`get_sync_committee_signatures` assumes `state` is the head state corresponding to processing the block at the current slot as determined by the fork choice (including any empty slots processed with `process_slots`), `validator_index` is the index of the validator in the registry `state.validators` controlled by `privkey`, and `privkey` is the BLS private key for the validator.
+
+```python
+def get_sync_committee_signatures(state: BeaconState, 
+                                  validator_index: ValidatorIndex, 
+                                  privkey: int) -> Sequence[SyncCommitteeSignature]:
+    sync_committee = state.current_sync_committee
+    pubkey = state.validators[validator_index].pubkey
+    sync_committee_indices = compute_indices_in_sync_committee(sync_committee, pubkey)
+
+    epoch = get_current_epoch(state)
+    domain = get_domain(state, DOMAIN_SYNC_COMMITTEE, epoch)
+    signing_root = compute_signing_root(get_block_root_at_slot(state, state.slot), domain)
+    signature = bls.Sign(privkey, signing_root)
+
+    return [
+        SyncCommitteeSignature(
+            slot=state.slot,
+            index=index,
+            signature=signature,
+        ) for index in sync_committee_indices
+    ]
+```
+
+##### Broadcast sync committee signature
+
+The validator broadcasts the assembled signature(s) to the assigned subnet(s), the `sync_committee_{subnet_id}` pubsub topic.
+
+The `subnet_id` is derived from the position in the sync committee such that the sync committee is divided into "subcommittees". It can be computed via `compute_subnet_for_sync_committee`:
+
+```python
+def compute_subnet_for_sync_committee(sync_committee_index: SyncCommitteeIndex) -> uint64:
+    return uint64(sync_committee_index % SYNC_COMMITTEE_SUBNET_COUNT)
```

Comment:
Will there be modulo bias towards low subnets? Or does it divide nicely?

Reply:
checking this out: https://research.kudelskisecurity.com/2020/07/28/the-definitive-guide-to-modulo-bias-and-how-to-avoid-it/, it seems that we are ok here as we are moving b/t powers of two (1024 % 8)